homeassistant/components/flux_led/__init__.py

```diff
@@ -19,14 +34,80 @@ async def async_setup(hass: HomeAssistant, config: dict):
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Set up Flux LED/MagicLight from a config entry."""
 
-    hass.async_create_task(
-        hass.config_entries.async_forward_entry_setup(entry, "light")
-    )
+    conf = entry.data
+    options = entry.options
+
+    if options.get("global", {}).get(CONF_AUTOMATIC_ADD, conf[CONF_AUTOMATIC_ADD]):
+        bulb_list = FluxLedList(
+            hass,
+            devices=conf[CONF_DEVICES],
+            config_entry=entry,
+        )
+
+        async def schedule_bulb_list_updates(_):
+            """Set up scheduled updates for the bulb list."""
+            await bulb_list.async_update()
+
+        async_track_time_interval(
+            hass,
+            schedule_bulb_list_updates,
+            timedelta(seconds=DEFAULT_NETWORK_SCAN_INTERVAL),
+        )
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
 
     return True
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class FluxLedList:
+    """Class to manage fetching the list of flux_led bulbs on the network."""
+
+    def __init__(self, hass: HomeAssistant, devices: dict, config_entry: ConfigEntry):
+        """Initialize the update manager."""
+
+        self.hass = hass
+        self._bulb_scan = BulbScanner()
+        self._config_entry = config_entry
+
+    @Throttle(timedelta(seconds=DEFAULT_NETWORK_SCAN_INTERVAL))
+    async def async_update(self):
+        """Fetch data from the network for flux_leds."""
+        bulb_list = await self.hass.async_add_executor_job(self._bulb_scan.scan)
+        known_bulbs = self._config_entry.data[CONF_DEVICES]
+
+        for bulb in bulb_list:
+            if bulb["ipaddr"].replace(".", "_") in known_bulbs:
+                continue
+
+            new_bulb_id = bulb["ipaddr"].replace(".", "_")
+            new_bulb = {
+                CONF_HOST: bulb["ipaddr"],
+                CONF_NAME: bulb["ipaddr"],
+            }
+
+            config_data = self._config_entry.data.copy()
+            config_data[CONF_DEVICES][new_bulb_id] = new_bulb
+            self.hass.config_entries.async_update_entry(
```

Comment:
Can we do one update of config entry data after all bulbs are checked?

Reply:
No problem - updated.